File: Matlab/+ACME/Color/normal2color.m
function [C] = normal2color(N)
% NORMAL2COLOR  Converts a given set of normals into a color vector
%   C = NORMAL2COLOR(N) returns a nx3 color vector from a nx3 matrix N.
C = normr(N);
C = (C + 1).*0.5;
end

File: encoding/normal2color.py
import torch


def normal2color(N):
    """
    Converts a given normal into a color

    Parameters
    ----------
    N : Tensor
        the input (n,3) normal tensor

    Returns
    -------
    Tensor
        the color coded normal
    """

    return torch.add(torch.mul(N, 0.5), 0.5)
